### ISSUE TITLE
Notifications and news

### ISSUE THREAD

**Issue by funilrys** [funilrys/funceble#11]
Hello world!

Just released the first version of **funceble**.

Wiki is on his way, but the script can be used out of the box without any big errors or bugs.

Big thanks to @mitchellkrogza && @PromoFaux :heart: 

P.S: Thanks to @pi-hole to for their big, big, biiiig list :wink: (I noticed that some of the domain in your lists are down or dead)

**Comment by PromoFaux** [comment]
Nice, just tried it out now. Interestingly if pi-hole is running all of the domains come back as `ACTIVE` (guessing it's because they are redirected to the Pi-hole itself!)

Our default lists are sourced from many maintainers (see [here for details](https://github.com/pi-hole/pi-hole/blob/master/adlists.default))

If nothing else, I'm inspired by your use of colour and formatting in the output of the scripts, might look into adding that to pi-hole!

**Comment by PromoFaux** [comment]
@wally3k, you may be interested in this project.

**Comment by funilrys** [comment]
@PromoFaux Interesting,
Maybe It's because of the logic.

Cause actually (It's only the first version), if we don't get data from the WHOIS Servers, we just execute `nslookup IPorDomain` and if we don't get `server can't find domain-name.me: NXDOMAIN` and specifically `NXDOMAIN` we just assign them to `ACTIVE`...

Thanks for the link added it to my work-flow for future features/tests.

**Comment by WaLLy3K** [comment]
Interesting! How long does it take to verify a list the size of Steven Blacks?

I host a ton of hosts files (1.6M!), and I'd be keen to check this out.

**Comment by funilrys** [comment]
@WaLLy3K I personally execute the script when I sleep so I can watch and analyze the result in the morning... So I can't tell now...

But thanks for the idea, gonna add an execution time result asap.
Thanks again.

**Comment by funilrys** [comment]
@WaLLy3K Ready in dev branch, gonna be in next release this morning/afternoon (Berlin Time)

**Comment by funilrys** [comment]
Just released [1.1.0](https://github.com/funilrys/funceble/releases/tag/1.1.0) !

**Comment by funilrys** [comment]
Just released [1.2.0](https://github.com/funilrys/funceble/releases/tag/1.2.0) ! 

@PromoFaux gonna test Pi-Hole list in the coming week or weekend :)

**Comment by mitchellkrogza** [comment]
Latest version working great @funilrys running a big test on an old + current list of mine and will then merge the two with active domains and be able to get rid of dead domains. Awesome tool !!!! 

PS - Doesn't run on macOS due to a sed issue after running `tool -i`

`sed: 1: "/Users/myusername/funceble/ ...": invalid command code m`

But I wouldn't bother with it, I am running it off my Ubuntu server overseas which is faster anyway.

**Comment by funilrys** [comment]
Just released [1.3.0](https://github.com/funilrys/funceble/releases/tag/1.3.0) !

Thanks @mitchellkrogza for #46 ! :+1: :100: :white_check_mark:

**Comment by mitchellkrogza** [comment]
Thanks @funilrys When running it now with latest version I get this error.

```
./funceble: line 322: expect: command not found
cat: /var/tmp/whois.32129: No such file or directory
```

**Comment by funilrys** [comment]
Just released [1.3.1](https://github.com/funilrys/funceble/releases/tag/1.3.1) !!

Thanks @mitchellkrogza  for #59  #58 ! :+1: :100:

**Comment by mitchellkrogza** [comment]
Thanks @funilrys been testing funceble inside travis to see if it can work in that environment.

When running ./tool -i I get an error in Travis

`/home/travis/build/mitchellkrogza/The-Big-List-of-Hacked-Malware-Web-Sites/travisCI/_funceble/tool: 148: /home/travis/build/mitchellkrogza/The-Big-List-of-Hacked-Malware-Web-Sites/travisCI/_funceble/tool: Syntax error: "(" unexpected (expecting "fi")`

and when running ./funceble -a -t 5 -h -f /home/travis/build/mitchellkrogza/The-Big-List-of-Hacked-Malware-Web-Sites/travisCI/_strip_domains/domains.txt I get an error (probably because tool -i failed)

`/home/travis/build/mitchellkrogza/The-Big-List-of-Hacked-Malware-Web-Sites/travisCI/_funceble/funceble: 139: /home/travis/build/mitchellkrogza/The-Big-List-of-Hacked-Malware-Web-Sites/travisCI/_funceble/funceble: Bad substitution`

You can see the build log here-https://travis-ci.org/mitchellkrogza/The-Big-List-of-Hacked-Malware-Web-Sites/builds/253548796

I've set my funceble executions in Travis to exit with 0 so it doesn't bring the repo commits down with failures. Interesting to play with it in Travis but seems Travis doesn't like some of the bash.

**Comment by funilrys** [comment]
@mitchellkrogza  Aw maybe I found where the problem is :stuck_out_tongue_closed_eyes: give a second gonna send a PR :)

**Comment by mitchellkrogza** [comment]
Thank @funilrys I love playing with Travis, but it can give you grey hairs. It sure can find the tiniest bugs or syntax errors in code.

**Comment by funilrys** [comment]
Just released [1.3.2](https://github.com/funilrys/funceble/releases/tag/1.3.2) :tada: :tada: !

**Comment by xxcriticxx** [comment]
@funilrys How do i run this with pihole on? All my domains show active like Adam said.

**Comment by mitchellkrogza** [comment]
@xxcriticxx  Keep something in mind here. When testing some of those lists which are subdomains of a main domain, they will all show as active. Take for instance https://github.com/funilrys/dead-hosts/blob/master/add.2o7Net/tested-list/add.2o7Net.list 

The domain 2o7.net is active so testing *.2o7.net will always return ACTIVE. 

You will also notice that on all those lists that specify hundreds of sub-domains you will never get and WHOIS value for EXPIRY DATE either as it cannot query that for a sub-domain only the main root domain.

This is one reason I suggested a feature or even a totally new project which can run funceble against these lists and produce a screenshot of each site in a nice PDF or HTML output which could help people seeing what sites are dead or still active.

In fact @funilrys if one could even do a simple curl and return the page title of each site it would quickly help people clean their lists. So an extra column showing the page title would quickly reveal those saying "Not Found" ?? Worth delving into this more.

**Comment by mitchellkrogza** [comment]
You will also notice any domain tested that includes a `www.` in front of it will show active because the root domain is active but will never return an expiry date from whois as it cannot query the expiry date for an A record.

**Comment by xxcriticxx** [comment]
What happens if domain is active but there is no page parked there. Would this script mark it as inactive?

**Comment by mitchellkrogza** [comment]
@xxcriticxx @funilrys it will show as ACTIVE. That's why a feature like a simple curl showing the page title in the results would give you something like this instead.

```
Domain                                                                                               Status      Expiration Date   Source     Analyse Date                      Page Title        
---------------------------------------------------------------------------------------------------- ----------- ----------------- ---------- -------------------------------   ------------------------------------------------------------------
7wind.ru                                                                                             ACTIVE      12-may-2018       WHOIS      Sat Jul 22 14:52:26 UTC 2017      404 Not Found
accentstudio.co.uk                                                                                   ACTIVE      28-dec-2017       WHOIS      Sat Jul 22 14:52:27 UTC 2017      Hello this web site has a Page Title
acenar.com                                                                                           ACTIVE      19-jan-2018       WHOIS      Sat Jul 22 14:52:27 UTC 2017      Hello this web page also has a page title
subdomain.acenar.com                                                                                 ACTIVE      19-jan-2018       WHOIS      Sat Jul 22 14:52:27 UTC 2017      404 Not Found
```

**Comment by xxcriticxx** [comment]
@mitchellkrogza can you run my personal list tell me of you get any inactive domains?

**Comment by mitchellkrogza** [comment]
@xxcriticxx where's the list? Send me link to the raw list.

**Comment by mitchellkrogza** [comment]
Is it this one? https://github.com/xxcriticxx/.pl-host-file/blob/master/hosts.txt

If so it's been tested already see: https://github.com/funilrys/dead-hosts/blob/master/pl-host-file/result.txt

**Comment by xxcriticxx** [comment]
got it thx guys

**Comment by mitchellkrogza** [comment]
Cool, @funlirys has done some awesome work with the dead hosts repo and of course let me say funceble is just the coolest tool ever. I hope @funilrys can add a column showing a page title in results.txt which would help immensely with picking up sites that no longer exist. Knowing @funilrys he's sitting tonight already pondering this :grin:

**Comment by mitchellkrogza** [comment]
You can see how I have implemented funceble on this repo and it's working like a charm https://github.com/mitchellkrogza/The-Big-List-of-Hacked-Malware-Web-Sites

**Comment by xxcriticxx** [comment]
i cant wait for next version 

@funilrys please chop chop lets make this best tool on the internet!

**Comment by mitchellkrogza** [comment]
+1 for that.

**Comment by funilrys** [comment]
@xxcriticxx Please avoid questions here, prefer to create a new issue.

**Comment by funilrys** [comment]
Just released [1.4.0](https://github.com/funilrys/funceble/releases/tag/1.4.0) !!

Thanks @mitchellkrogza  for #64  ! :+1: :100: 
Thanks @xxcriticxx   for #67 #69 ! :+1: :100:

**Comment by funilrys** [comment]
Hello everyone (+ @SMed79) I have some short information and request!

Before anything, I want to thank all of the people reading this for using, contributing, testing or simply supporting me and/or this project in the past, the present or the future. I know that some of you are waiting for the next release which was supposed to be out months ago but I changed my plan but I'll do my best to release it as soon as possible. 

The current state of the `dev` branch of Funceble is ready for release as it was supposed to be but I wanted to release Funceble (with some more improvement) at the same time with its recently born little sister [PyFunceble](https://github.com/funilrys/PyFunceble/tree/dev).
PyFunceble is quite equivalent to Funceble but written in Python and with many code improvements which, as said previously, are coming as soon as possible into Funceble.

The reason I'm writing this is that I wanted to know if some of you can or would like to help break PyFunceble so that PyFunceble (and lately Funceble) will become better!
Interested ? Clone [PyFunceble](https://github.com/funilrys/PyFunceble/tree/dev) break it and publish the discovered issue(s) [over here](https://git.io/vbBBA)! 

Thanks again for using, contributing, testing or simply supporting me and/or this project.
I wish you all a great weekend!

Cheers.
